esx_service/utils/kvESX.py

```diff
@@ -166,35 +168,45 @@ def delete(volpath):
     return True
 
 
+# Align a given string to the specified block boundary.
+def align_str(kv_str, block):
+   # Align string to the next block boundary.
+   new_len = int((len(kv_str) + block - 1) / block) * block - 1
+   padded_str = '{:<{width}}'.format(kv_str, width=new_len)
+
+   # Add a trailing newline to bring up string length to a 4K boundary.
+   return '{0}{1}'.format(padded_str, '\n')
+
 # Load and return dictionary from the sidecar
 def load(volpath):
     metaFile = lib.DiskLib_SidecarMakeFileName(volpath, DVOL_KEY)
 
     try:
-        fh = open(metaFile, "r+")
-    except IOError:
-        logging.exception("Failed to open %s", metaFile);
+       with open(metaFile, "r+") as fh:
+          kv_str = fh.read()
+    except:
+        logging.exception("Failed to access %s", metaFile);
         return None
 
-    kvDict = json.load(fh)
-
-    fh.close()
-
-    return kvDict
+    try:
+       return json.loads(kv_str)
+    except ValueError:
+       logging.exception("Failed to decode meta-data for %s", volpath);
+       return None
 
 
 # Save the dictionary to side car.
-def save(volpath, kvDict):
+def save(volpath, kv_dict):
     metaFile = lib.DiskLib_SidecarMakeFileName(volpath, DVOL_KEY)
 
+    kv_str = json.dumps(kv_dict)
+
     try:
-        fh = open(metaFile, "w+")
-    except IOError:
-        logging.exception("Failed to open %s", metaFile);
+       with open(metaFile, "w+") as fh:
```

Comment:
For both r+ and w+ this is left over from earlier when the code was being developed and was reading and updating the meta-data in the same functions. Its not needed.